homeassistant/components/netgear/__init__.py

```diff
@@ -1 +1,75 @@
-"""The netgear component."""
+"""Support for Netgear routers."""
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import CONF_TRACKED_LIST, DOMAIN, PLATFORMS
+from .router import NetgearRouter, convert_tracked_list
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up Netgear component."""
+    router = NetgearRouter(hass, entry)
+    await router.async_setup()
```

Comment:
Trap retry-able exceptions and throw `ConfigEntryNotReady`

Trap authentication exceptions and throw `ConfigEntryAuthFailed` if you have implemented reauth.  Otherwise, logger the error and return False

Reply:
reath is not implemented and that is for a future PR.
Login errors are not distinguised, so I will just throw ConfigEntryNotReady, in a future PR the upstream lib can be changed to seperate wrong credentials and network timeouts etc.